src/Analyzers/Orchestration/DateTimeOrchestrationAnalyzer.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Concurrent;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+using Microsoft.CodeAnalysis;
+using Microsoft.DurableTask.Analyzers.Helpers;
+
+namespace Microsoft.DurableTask.Analyzers.Orchestration;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+public sealed class DateTimeOrchestrationAnalyzer : OrchestrationAnalyzer
```

Comment:
Will this also flag usage of `ITimeProvider`? Or will that be a separate analyzer?

Reply:
I believe it can be a different diagnostic/analyzer. I created an issue so we can track it: https://github.com/microsoft/durabletask-dotnet/issues/285